server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -151,32 +236,51 @@ def mock_results_push(self, ctrl, tb, token):
 
     def test_multiple_process_tb(self, monkeypatch, caplog, test_logger):
         """verify tar balls processing at the time of Failure as well as success"""
-        tb_path = fixture_path / "mix-tarball"
-        good_tb = tb_path / tarball
-        bad_tb = tb_path / bad_tarball
-
-        def mock_remove(path, *, dir_fd=None):
-            assert path in [good_tb, bad_tb]
+        tb_path = fixture_path
+        receive_path = Path(str(tb_path.resolve()) + "-002")
+        good_tb = receive_path / tarball
+        bad_tb = receive_path / bad_tarball
 
         def mock_get(self, section: str, option: str):
-            assert option in ["pbench-receive-dir-prefix", "put-token"]
+            assert option in [
+                "pbench-receive-dir-prefix",
+                "put-token",
+            ], f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
             return (
                 tb_path
                 if option == "pbench-receive-dir-prefix"
                 else "Authorization-token"
             )
 
+        def mock_glob(path, pattern):
+            tbmd5_1 = path / "log.tar.xz.md5"
+            tbmd5_2 = path / "bad_log.tar.xz.md5"
+            if path == receive_path and pattern == "**/*.tar.xz.md5":
+                assert (
+                    tbmd5_1.suffix == ".md5" and tbmd5_1.suffix == ".md5"
+                ), f"Unexpected tar ball, {path!r} in mocked Path.glob()"
+                return [tbmd5_1, tbmd5_2]
+
         def mock_results_push(self, ctrl, tb, token):
-            bad_tb = self.receive_dir / bad_tarball
-            if bad_tb == tb:
+            if tb == bad_tb:
                 raise FileNotFoundError(f"No such file or directory: '{tb.name}'")
             return
 
-        expected_result = Results(nstatus=f": processed {tarball}\n", ntotal=2, ntbs=1,)
+        def mock_remove(path, *, dir_fd=None):
+            assert path in [
+                good_tb,
+                Path(str(good_tb) + ".md5"),
+                bad_tb,
+            ], f"Unexpected tar ball, {path!r} in mocked os.remove()"
```

Comment:
The `bad_tb` has no `.md5` file?

Reply:
I am confused. considering the normal flow I believe it should not go beyond the `results_push` function, So, I will go ahead with it and remove any reference to `bad_tb`. What do you think?

Reply:
I think it might be helpful for both you as developer and me as reviewer if you were to put a docstring on this test which describes what you intend it to do, step by step, with some detail.

For instance:
- we expect the CUT to invoke the mocked config `get()` function, which will provide it with the path to the tarballs and MD5 files
- we expect the path returned by `get()` to be passed to the mocked `Path.glob()` function which will return MD5 files for a "bad" and a "good" tarball, respectively
- we expect those two paths to be passed in successive calls to the mocked `results_push()` function; it will raise a file-not-found exception for the "bad" one and it will return normally for the "good" one
- we expect the CUT to catch the exception and to proceed to the next file path
- we expect that the mocked `os.remove()` function will be invoked for the good tarball and its MD5 file, but not for the bad tarball or its MD5 file.
- and, we expect that the CUT will report two tarballs processed, one successfully and one with errors.

This exercise should help make the flow clear in your head and guide the reader through what you've written.  (It's kind of like producing a specification for what you're building; ideally, the code would provide this for us, but, in a unit test with mocks, it's not obvious without consulting the CUT what the flow is supposed to be.)

Reply:
OK, so I didn't really answer your question....

I think you are correct:  because mocked `results_push()` raised an exception, the mocked `os.remove()` is never called for either of the "bad" tar ball files.  So, we don't need (or want) to include them in the assertion in the mock.  (So, we were both right!  😉  The CUT should not be trying to remove the MD5 file for the bad tarball, and it should not be trying to remove the bad tarball.  😁)